soc/xtensa/nxp_adsp/imx8/include/_soc_inthandlers.h

```diff
@@ -0,0 +1,282 @@
+
+uint32_t my_int_mask_3_cnt_logger_started = 0;
+uint32_t my_int_mask_3_cnt_work_q_started = 0;
+
+uint32_t logger_enabled_mask_3_cnt_is;
+uint32_t work_q_enabled_mask_3_cnt_is;
+
+uint32_t logger_started_mask_3_cnt_is;
+uint32_t work_q_started_mask_3_cnt_is;
+
+extern uint32_t logger_enabled;
+extern uint32_t work_q_enabled;
+
+extern uint32_t logger_started;
+extern uint32_t work_q_started;
+
+extern uint32_t sof_status1[3];
+extern uint32_t sof_status2[3];
+extern uint32_t sof_status3[3];
+
+uint32_t my_int_mask_1[100];
+uint32_t my_int_mask_1_cnt = 0;
+
+uint32_t my_int_mask_2[100];
+uint32_t my_int_mask_2_cnt = 0;
+
+uint32_t my_int_mask_5[100];
+uint32_t my_int_mask_5_cnt = 0;
+
+static inline int _xtensa_handle_one_int1(unsigned int mask)
+{
+	int irq;
+
+	if (my_int_mask_1_cnt < 100)
+		my_int_mask_2[my_int_mask_1_cnt] = mask;
+
+	my_int_mask_1_cnt++;
+
+	if (mask & BIT(8)) {
+		mask = BIT(8);
+		irq = 8;
+		goto handle_irq;
+	}
+	return 0;
+handle_irq:
+	_sw_isr_table[irq].isr(_sw_isr_table[irq].arg);
+	return mask;
+}
+
+static inline int _xtensa_handle_one_int2(unsigned int mask)
+{
+	int irq;
+	int i = 0;
+
+	if (my_int_mask_2_cnt < 100)
+		my_int_mask_2[my_int_mask_2_cnt] = mask;
+
+	my_int_mask_2_cnt++;
+
+	mask &= XCHAL_INTLEVEL2_MASK;
+	for (i = 0; i <= 31; i++)
+		if (mask & BIT(i)) {
+			mask = BIT(i);
+			irq = i;
+			goto handle_irq;
+		}
+	return 0;
+handle_irq:
+	_sw_isr_table[irq].isr(_sw_isr_table[irq].arg);
+	return mask;
+}
+
+static inline int _xtensa_handle_one_int3(unsigned int mask)
+{
+	int irq;
+
+	if (my_int_mask_3_cnt < 150)
+		my_int_mask_3[my_int_mask_3_cnt] = mask;
+
+	my_int_mask_3_cnt++;
+
+	if (logger_enabled) {
+		my_int_mask_3_cnt_logger_enabled++;
+		if (my_int_mask_3_cnt_logger_enabled == 1)
+			// save and display my_int_mask_3_cnt
+			logger_enabled_mask_3_cnt_is = my_int_mask_3_cnt;
+	}
+
+	if (logger_started) {
+		my_int_mask_3_cnt_logger_started++;
+		if (my_int_mask_3_cnt_logger_started == 1)
+			// save and display my_int_mask_3_cnt
+			logger_started_mask_3_cnt_is = my_int_mask_3_cnt;
+	}
+
+	if (work_q_enabled) {
+		my_int_mask_3_cnt_work_q_enabled++;
+		if (my_int_mask_3_cnt_work_q_enabled == 1)
+			// save and display my_int_mask_3_cnt
+			work_q_enabled_mask_3_cnt_is = my_int_mask_3_cnt;
+	}
+
+	if (work_q_started) {
+		my_int_mask_3_cnt_work_q_started++;
+		if (my_int_mask_3_cnt_work_q_started == 1)
+			// save and display my_int_mask_3_cnt
+			work_q_started_mask_3_cnt_is = my_int_mask_3_cnt;
+	}
+
+	if (mask & BIT(1)) {
+		mask = BIT(1);
+		irq = 1;
+		goto handle_irq;
+	}
+	if (mask & BIT(3)) {
+		mask = BIT(3);
+		irq = 3;
+		goto handle_irq;
+	}
+	if (mask & BIT(31)) {
+		mask = BIT(31);
+		irq = 31;
+		goto handle_irq;
+	}
+	return 0;
+handle_irq:
+	_sw_isr_table[irq].isr(_sw_isr_table[irq].arg);
+
+	my_cache_flush(sof_status1, sizeof(sof_status1));
+	my_cache_flush(sof_status2, sizeof(sof_status2));
+	my_cache_flush(sof_status3, sizeof(sof_status3));
```

Comment:
I'm guessing again that this is debug code?  Obviously that warning about about automatically generated code is being violated otherwise (though we never integrated that python script into the proper build and IIRC it's been accidentally removed from the tree a few times... really Xtensa interrupt entry needs a rewrite)

Reply:
Yes, also debug code.

Reply:
I've removed all the debug code.